showcase/go/showcase_integration_test.go

```diff
@@ -0,0 +1,200 @@
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	conn, err := grpc.Dial("localhost:7469", grpc.WithInsecure())
+	if err != nil {
+		log.Fatal(err)
+	}
+	clientOpt := option.WithGRPCConn(conn)
+	client, err = showcase.NewEchoClient(context.Background(), clientOpt)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestEcho(t *testing.T) {
+	content := "hello world!"
+	req := &genprotopb.EchoRequest{
+		Response: &genprotopb.EchoRequest_Content{
+			Content: content,
+		},
+	}
+	resp, err := client.Echo(context.Background(), req)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.GetContent() != req.GetContent() {
+		t.Errorf(
+			"Echo did not receive expected value. Got %s, Wanted %s",
+			resp.GetContent(),
+			req.GetContent())
+	}
+}
+
+func TestEcho_error(t *testing.T) {
+	val := codes.Canceled
+	req := &genprotopb.EchoRequest{
+		Response: &genprotopb.EchoRequest_Error{
+			Error: &spb.Status{Code: int32(val)},
+		},
+	}
+	_, err := client.Echo(context.Background(), req)
+
+	if err == nil {
+		t.Errorf("Echo called with code %d did not return an error.", val)
+	}
+	status, _ := status.FromError(err)
+	if status.Code() != val {
+		t.Errorf("Echo called with code %d returned an error with code %d", val, status.Code())
+	}
+}
+
+func TestExpand(t *testing.T) {
+	content := "The rain in Spain stays mainly on the plain!"
+	req := &genprotopb.ExpandRequest{Content: content}
+	s, err := client.Expand(context.Background(), req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	resps := []string{}
+	for {
+		resp, err := s.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		resps = append(resps, resp.GetContent())
+	}
+	got := strings.Join(resps, " ")
+	if content != got {
+		t.Errorf("Expand expected %s but got %s", content, got)
+	}
+}
+
+func TestCollect(t *testing.T) {
+	content := "The rain in Spain stays mainly on the plain!"
+	s, err := client.Collect(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, str := range strings.Split(content, " ") {
+		s.Send(&genprotopb.EchoRequest{
+			Response: &genprotopb.EchoRequest_Content{Content: str}})
+	}
+
+	resp, err := s.CloseAndRecv()
+	if content != resp.GetContent() {
+		t.Errorf("Collect failed, expected %s, got %s", content, resp.GetContent())
+	}
+}
+
+func TestChat(t *testing.T) {
+	content := "The rain in Spain stays mainly on the plain!"
+	s, err := client.Chat(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, str := range strings.Split(content, " ") {
+		s.Send(&genprotopb.EchoRequest{
+			Response: &genprotopb.EchoRequest_Content{Content: str}})
+	}
+	s.CloseSend()
+	resps := []string{}
+	for {
+		resp, err := s.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		resps = append(resps, resp.GetContent())
+	}
+	got := strings.Join(resps, " ")
+	if content != got {
+		t.Errorf("Chat expected %s but got %s", content, got)
+	}
+}
+
+func TestWait(t *testing.T) {
```

Comment:
Do we need to test the case with timeout as well?

Reply:
Done!